arlas-core/src/main/java/io/arlas/server/core/FluidSearch.java

```diff
@@ -776,45 +777,43 @@ else if (aggregationBuilder instanceof GeoGridAggregationBuilder)
+                } else if (ag == AggregatedGeometryEnum.CENTROID) {
+                    ValuesSourceAggregationBuilder metricAggregation = AggregationBuilders.geoCentroid(AggregatedGeometryEnum.CENTROID.value() + AGGREGATED_GEOMETRY_SUFFIX).field(aggregationGeoField);
                     aggregationBuilder.subAggregation(metricAggregation);
                 }
-            } else if (aggregationModel.fetchGeometry.strategy == AggregatedGeometryStrategyEnum.geohash) {
-                // aggregationModel.type is necesseraly AggregationTypeEnum.geohash. We already return the centroid of each geohash by default => nothing to implement here => create geohash geometry at response stage
-            } else if (aggregationModel.fetchGeometry.strategy == AggregatedGeometryStrategyEnum.byDefault) {
-                if (aggregationModel.type !=  AggregationTypeEnum.geohash) {
-                    String[] includes = {collectionReference.params.geometryPath, collectionReference.params.centroidPath};
-                    TopHitsAggregationBuilder topHitsAggregationBuilder = AggregationBuilders.topHits(RANDOM_GEOMETRY).size(1).fetchSource(includes, null);
-                    aggregationBuilder.subAggregation(topHitsAggregationBuilder);
-                }
-                // if aggregationModel.type ==  AggregationTypeEnum.geohash then we already return the centroid of each geohash by default => nothing to implement
-            } else {
-                String[] includes = {collectionReference.params.geometryPath, collectionReference.params.centroidPath};
-                String sortField = (aggregationModel.fetchGeometry.field != null) ? aggregationModel.fetchGeometry.field : collectionReference.params.timestampPath;
-                if (aggregationModel.fetchGeometry.strategy == AggregatedGeometryStrategyEnum.first) {
-                    TopHitsAggregationBuilder topHitsAggregationBuilder = AggregationBuilders.topHits(FIRST_GEOMETRY).size(1).sort(sortField, SortOrder.ASC).fetchSource(includes, null);
-                    aggregationBuilder.subAggregation(topHitsAggregationBuilder);
-                } else if (aggregationModel.fetchGeometry.strategy == AggregatedGeometryStrategyEnum.last) {
-                    TopHitsAggregationBuilder topHitsAggregationBuilder = AggregationBuilders.topHits(LAST_GEOMETRY).size(1).sort(sortField, SortOrder.DESC).fetchSource(includes, null);
-                    aggregationBuilder.subAggregation(topHitsAggregationBuilder);
+            });
+        }
+        return aggregationBuilder;
+    }
+
+    private ValuesSourceAggregationBuilder setRawGeometries(Aggregation aggregationModel, ValuesSourceAggregationBuilder aggregationBuilder) throws ArlasException {
+        if (aggregationModel.rawGeometries != null) {
+            if (!aggregationModel.rawGeometries.geometries.isEmpty()) {
+                String[] includes = aggregationModel.rawGeometries.geometries.stream().toArray(String[]::new);
```

Comment:
`aggregationModel.rawGeometries.geometries` is already a list, I need to transform it to a array because the elasticsearch methd `fetchSource` needs this format